src/util/stl-utils.h

```diff
@@ -292,6 +292,29 @@ struct CompareFirstMemberOfPair {
   }
 };
 
+/// Comparator object that does comparison on a different
```

Comment:
This comparator is not very well explained.
However, I think I see what you are trying to do here-- it would be better, instead of using this comparator, to instead sort a vector of pairs, where the 1st of each pair is the thing you want to sort on, and the 2nd is the index that you want to sort.